### ISSUE TITLE
[HOLD for payment 2025-01-22] [$250] Search - No result when searching for current email address and group chat

### ISSUE THREAD

**Issue by mitarachim** [Expensify/App#54594]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.79-0
**Reproducible in staging?:** Yes
**Reproducible in production?:** No
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** Exp
**Email or phone of affected tester (no customers):** applausetester+8078787@applause.expensifail.com
**Issue reported by:** Applause Internal Team
**Device used:** Mac 15.0 / Chrome
**App Component:** Search

## Action Performed:
Precondition:
- User is logged in to Expensifail account (many contacts).

1. Go to staging.new.expensify.com
2. Open search router.
3. Enter current email address.
4. Close the search router.
5. Open FAB > Start chat.
6. Create a group chat with a few users.
7. Open search router.
8. Enter the group name.

## Expected Result:
Self DM and group chat will appear in the search result.

## Actual Result:
There is no result when searching for current email address and group chat.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [x] iOS: Standalone
- [x] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/b7fabe01-c249-45a1-9915-bd271203c437

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021872596042643273347</li>
        <li>Upwork Job ID: 1872596042643273347</li>
        <li>Last Price Increase: 2024-12-27</li>
<li>Automatic offers: </li>
<ul>
<li>allgandalf | Reviewer | 105503742</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @laurenreidexpensify</details>

**Comment by Beamanator** [comment]
@hannojg it looks like this could be related to https://github.com/Expensify/App/pull/51954, can you please have a look? 🙏

**Comment by Beamanator** [comment]
Marking external b/c i'm 99% sure this is related to the above linked PR

**Comment by jasperhuangg** [comment]
@hannojg Please take a look when you can!

**Comment by hannojg** [comment]
Sorry, was OOO - looking into this now!!!

**Comment by hannojg** [comment]
Okay, I can't reproduce the bug on latest staging or main branch:

https://github.com/user-attachments/assets/a2b5b303-aecc-4a3b-bdcb-be049b76b556

Will run a bisect to see if I can find the faulty commit

**Comment by hannojg** [comment]
Hm, can't reproduce it on the commit for version `9.0.79-0` either where it was originally reproduced 🤔

**Comment by hannojg** [comment]
https://github.com/user-attachments/assets/b8678edc-5dc4-4fa1-b9cb-19284528d848

**Comment by hannojg** [comment]
Not sure, I am not able to reproduce this in any way. Can we please try to reproduce on latest staging @mitarachim ?

**Comment by Beamanator** [comment]
aah interesting, thanks so much for checking @hannojg ! I will try as well, if I also can't reproduce, let's close this out 👍

**Comment by Beamanator** [comment]
Yeah same, I believe I followed the same steps & was able to find current email address & group chats in the search router 🤷  so let's close this out!

**Comment by mitarachim** [comment]
tester not available to retest.
i can reproduce followed steps using Expensifail account (many contacts).
Staging build v9.0.79-2

https://github.com/user-attachments/assets/2995acc2-2ad6-44d7-a28b-9fbe070afd10

**Comment by hannojg** [comment]
Hm very odd, thanks for retesting. Is there any way how I could access these accounts where it's failing (supportal access? we can sync on that on slack).
I can't reproduce it with a high traffic account (which also has lots of contacts)

**Comment by mitarachim** [comment]
@hannojg try with my account applausetester+mr101604@applause.expensifail.com

**Comment by hannojg** [comment]
What is your slack name? (Will reach out to you for getting the magic code)

**Comment by mitarachim** [comment]
my slack name : Mita rachim

**Comment by hannojg** [comment]
I can reproduce on Mita's account, looking into this now!

**Comment by hannojg** [comment]
Yes, the above assumption is correct. The data for Mita's account is `639,337` chars long, and we currently have a hard limit of `400,000`. Will write a fix for this to be dynamic based on this issue which should fix this issue as well:

- https://github.com/Expensify/App/issues/51151

**Comment by Beamanator** [comment]
@hannojg thanks for continuing to look into this, do you think this needs to be a blocker again?

**Comment by hannojg** [comment]
It only is going to happen for super big accounts, not sure how many real customer accounts have over 12k personal details?

**Comment by Beamanator** [comment]
Thaaaaaat's very fair 😅  i know we "do" / "will" have mega customers on NewDot, but atm i think unlikelyyyy

**Comment by hannojg** [comment]
PR is ready for review:

- https://github.com/Expensify/App/pull/54656

**Comment by laurenreidexpensify** [comment]
Payment Summary: 

C+ reviewer: @allgandalf paid $250 in Upwork 👍
